### DISCUSSION TITLE
[RFC] Yet another buffer usage API revision.

### DISCUSSION THREAD

**Discussion by iwoplaza** [software-mansion/TypeGPU#693] [Ideas]
I propose we change the buffer usage extraction API from `.as*` or `as*(...)` to `.as('*')`.

An example:
```ts
const fooBuffer = root
  .createBuffer(d.f32, 123)
  .$usage('uniform', 'storage');

const fooUniform = fooBuffer.as('uniform');
const fooMutable = fooBuffer.as('mutable');
const fooReadonly = fooBuffer.as('readonly');
```

Benefits:
- We can safely type the input to the `as` method, rather than having to change the output based on available flags.

**Comment by mhawryluk** [comment]
I like that this way it is more consistent with the `.asView` api for textures

**Comment by mhawryluk** [comment]
done in #836
